### PR TITLE
invites table migration

### PR DESCRIPTION
**Pull request by alishakawaguchi**

### CODE REVIEW

File: backend/hack/migration-add.sh [hunk 1]
```diff
@@ -14,4 +14,4 @@ SCRIPT_DIR=$(cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd)
 
 # use epoch time so unlikely to clash
 set -v
-migrate create -ext sql -dir "${SCRIPT_DIR}/../migrations" "$1"
+migrate create -ext sql -dir "${SCRIPT_DIR}/../sql/postgresql/schema" "$1"
```
Comment (nickzelei) [inline backend/hack/migration-add.sh:17]: whoop sorry, missed this ref update.

File: backend/sql/postgresql/schema/20231108001411_invites.up.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  email varchar NOT NULL,
+  token varchar NOT NULL DEFAULT gen_random_uuid(),
+  accepted boolean DEFAULT false,
+	created_at timestamp NOT NULL DEFAULT now(),
+	updated_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  expires_at timestamp NOT NULL,
+	CONSTRAINT invites_pkey PRIMARY KEY (id),
+  CONSTRAINT fk_invites_accounts_id FOREIGN KEY (account_id) REFERENCES neosync_api.accounts(id) ON DELETE CASCADE,
+  CONSTRAINT fk_invites_users_id FOREIGN KEY (id) REFERENCES neosync_api.users(id) ON DELETE CASCADE,
+  CONSTRAINT invites_email_account_id UNIQUE (email, account_id)
```
Comment (nickzelei) [inline backend/sql/postgresql/schema/20231108001411_invites.up.sql:14]: I don't know if we want this constraint. If an email gives invited but the invitation expires, they should be able to receive another invite without having to delete the previous one.

File: backend/sql/postgresql/schema/20231108001411_invites.up.sql [hunk 3]
```diff
@@ -0,0 +1,19 @@
+CREATE TABLE IF NOT EXISTS neosync_api.invites (
+	id uuid NOT NULL DEFAULT gen_random_uuid(),
+  account_id uuid NOT NULL,
+  sender_user_id uuid NOT NULL,
```
Comment (nickzelei) [inline backend/sql/postgresql/schema/20231108001411_invites.up.sql:4]: I'm thinking this should be NULL and the FK constraint should set to null if the user_id is deleted. That way a user could be removed, but the invite is retained.
Any thoughts?

Comment (alishakawaguchi) [reply backend/sql/postgresql/schema/20231108001411_invites.up.sql]: yeah that makes sense

**Review by nickzelei — APPROVED** [review]
🚢

**Pull request merged**